ospfd/ospf_dump.c

```diff
@@ -1457,194 +1457,221 @@ DEFUN (no_debug_ospf_instance_nssa,
 	return CMD_SUCCESS;
 }
 
-DEFUN (debug_ospf_te,
+DEFPY (debug_ospf_te,
        debug_ospf_te_cmd,
-       "debug ospf te",
-       DEBUG_STR
-       OSPF_STR
-       "OSPF-TE information\n")
-{
-	if (vty->node == CONFIG_NODE)
-		CONF_DEBUG_ON(te, TE);
-	TERM_DEBUG_ON(te, TE);
-	return CMD_SUCCESS;
-}
-
-DEFUN (no_debug_ospf_te,
-       no_debug_ospf_te_cmd,
-       "no debug ospf te",
+       "[no$no] debug ospf [(1-65535)$instance] te",
        NO_STR
        DEBUG_STR
        OSPF_STR
+       "Instance ID\n"
        "OSPF-TE information\n")
 {
-	if (vty->node == CONFIG_NODE)
-		CONF_DEBUG_OFF(te, TE);
-	TERM_DEBUG_OFF(te, TE);
+	if (instance && instance != ospf_instance)
+		return CMD_NOT_MY_INSTANCE;
+
+	if (vty->node == CONFIG_NODE) {
+		if (no)
+			DEBUG_OFF(te, TE);
```

Comment:
Shouldn't we use CONF_DEBUG_OFF/CONF_DEBUG_ON if CONFIG_NODE?

Reply:
Hi,
if we use only CONF_DEBUG_OFF/CONF_DEBUG_ON, debug command is written to running config but debug flag is not set.
hence logs are not printed.
so,

we can use like 
if (vty->node == CONFIG_NODE) {
     DEBUG_OFF/DEBUG_ON()
} else {
     TERM_DEBUG_ON/TERM_DEBUG_OFF()
}

or

if (vty->node == CONFIG_NODE) {
     CONF_DEBUG_OFF/CONF_DEBUG_ON()
}
TERM_DEBUG_ON/TERM_DEBUG_OFF()

Reply:
ospfd.rst is not updated with commands like debug ospf sr/ti-lfa etc
and the command should be like 
debug ospf [(1-65535)$instance] zebra
debug ospf [(1-65535)$instance] te

shall I update the doc in this PR itself or update it in new PR ?

Reply:
>shall I update the doc in this PR itself or update it in new PR ?

You can do it here.

Reply:
Sure